kernel/init.c

```diff
@@ -427,7 +425,10 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 #if defined(CONFIG_STACK_POINTER_RANDOM) && (CONFIG_STACK_POINTER_RANDOM != 0)
 	z_stack_adjust_initialized = 1;
 #endif /* CONFIG_STACK_POINTER_RANDOM */
+
+#if defined(CONFIG_BOOT_BANNER)
```

Comment:
boot_banner should also run when `CONFIG_BOOT_BANNER=n` to handle the delayed boot option which does not depend on the boot banner.